### PR TITLE
Stats Subscribers: Add Period Navigation

### LINKED ISSUE #75002: Stats: Add period selector to the subscribers chart
Subscribers chart API will return data for a specific period, similar to the visitors' chart.
We need to extend the chart with a period selector to accommodate it.

Development tasks:
- [x] add a period selection component used for the visitors' chart (create a new component if one doesn't exist to avoid repeating the code)
- [ ] investigate how different periods are stored in Redux for the visitors' chart
- [ ] create a new issue to update the implementation if it's needed as a result of the above investigation

### PR DESCRIPTION
**Pull request by annacmc**
Related to https://github.com/Automattic/wp-calypso/issues/75002

## Proposed Changes

* This PR adds period navigation to the subscriber's chart. It is not currently fully functional, as the chart will need some additional changes (to use real data) in order to update visually based on the URL query string. However the period navigation should correctly update the URL query string. 

## Testing Instructions

* Open the Calypso Live Branch. 
* Navigate to `/stats/subscribers/{siteURL}?flags=stats/subscribers-section`
* Check the period navigation buttons (day/week/month/year etc) 

image

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/my-sites/stats/subscribers-section/index.tsx [hunk 1]
```diff
@@ -29,8 +33,15 @@ function transformData( data: SubscribersData[] ): uPlot.AlignedData {
 	return [ x, y ];
 }
 
-export default function SubscribersSection( { siteId }: { siteId: string } ) {
-	const period = 'month';
+export default function SubscribersSection( {
+	siteId,
```
Comment (grzegorz-cp) [inline client/my-sites/stats/subscribers-section/index.tsx]: After renaming this to `slug` nothing is passing `siteId`. Let's remove it.

File: client/my-sites/stats/subscribers-section/index.tsx [hunk 2]
```diff
@@ -58,9 +79,19 @@ export default function SubscribersSection( { siteId }: { siteId: string } ) {
...
 			{ ! isLoading && chartData.length === 0 && (
-				<p className="subscribers-section__no-data">No data availble for the specified period.</p>
+				<p className="subscribers-section__no-data">No data available for the specified period.</p>
 			) }
 			{ errorMessage && <div>Error: { errorMessage }</div> }
+			{ ! isLoading && (
+				<div>
+					<StatsPeriodHeader>
+						<StatsPeriodNavigation date={ date } period={ period } url={ pathTemplate }>
+							<DatePicker period={ period } date={ date } statsType="statsTopPosts" showQueryDate />
```
Comment (grzegorz-cp) [inline client/my-sites/stats/subscribers-section/index.tsx]: Can you rename or remove `statsTopPosts`, or does it break something at the moment?

File: client/my-sites/stats/subscribers-section/index.tsx [hunk 3]
```diff
@@ -50,15 +61,40 @@ export default function SubscribersSection( { siteId }: { siteId: string } ) {
...
 			<div className="subscribers-section-heading">
 				<h1 className="highlight-cards-heading">Subscribers</h1>
+				{ ! isLoading && (
+					<div>
+						<StatsPeriodHeader>
+							<StatsPeriodNavigation date={ date } period={ period } url={ pathTemplate }>
+								<DatePicker
+									period={ period }
+									date={ date }
+									statsType="statsTopPosts"
```
Comment (grzegorz-cp) [inline client/my-sites/stats/subscribers-section/index.tsx]: Can we rename or remove `statsTopPosts`, or does it break something at the moment?

**Review by grzegorz-cp — CHANGES_REQUESTED** [review]
Thanks for fixing the redirect to the traffic page when pressing the period buttons.
I moved the legend to be closer to the chart and left two requests. I think after this the PR will be in a good shape.

**Review by grzegorz-cp — APPROVED** [review]
Looks good, but let's have one more pair of eyes look at it.

**Comment by annacmc** [comment]
> Looks good, but let's have one more pair of eyes look at it.

Thanks @grzegorz-cp I've published the PR now, however, noticing an eslint check is failing since removing `statsTopPosts`  Going to look into it in the morning. Thought I'd get this out for review in the meantime.

File: client/my-sites/stats/subscribers-section/index.tsx [hunk 4]
```diff
@@ -58,28 +61,35 @@ export default function SubscribersSection( {
...
+					</div>
+				) }
 				<div className="subscribers-section-legend" ref={ legendRef }></div>
 			</div>
 			{ isLoading && <StatsModulePlaceholder className="is-chart" isLoading /> }
 			{ ! isLoading && chartData.length === 0 && (
-				<p className="subscribers-section__no-data">
-					{ translate( 'No data availble for the specified period.' ) }
-				</p>
+				<p className="subscribers-section__no-data">No data available for the specified period.</p>
```
Comment (dognose24) [inline client/my-sites/stats/subscribers-section/index.tsx]: Don't we need the i18n here anymore?

File: client/my-sites/stats/subscribers-section/index.tsx [hunk 5]
```diff
@@ -58,28 +61,35 @@ export default function SubscribersSection( {
...
-						<a
-							className="highlight-cards-heading-wrapper"
-							href={ '/people/subscribers/' + siteSlug }
-						>
-							{ translate( 'View all subscribers' ) }
-						</a>
-					</small>
-				</h1>
+			<div className="subscribers-section-heading">
+				<h1 className="highlight-cards-heading">Subscribers</h1>
```
Comment (dognose24) [inline client/my-sites/stats/subscribers-section/index.tsx]: I noticed the removed code was just [added by the PR](https://github.com/Automattic/wp-calypso/pull/75653) a few hours ago. Maybe the regression is due to merging the trunk.

Comment (jsnmoon) [reply client/my-sites/stats/subscribers-section/index.tsx]: Agreed, this appears to unintentionally revert #75653.

File: client/my-sites/stats/subscribers-section/index.tsx (hunk 4 above)
Comment (jsnmoon) [reply client/my-sites/stats/subscribers-section/index.tsx]: This also appears to be an unintentional revert of #75653.

File: client/my-sites/stats/subscribers-section/index.tsx [hunk 6]
```diff
@@ -58,28 +61,35 @@ export default function SubscribersSection( {
...
+				{ ! isLoading && (
+					<div>
+						<StatsPeriodHeader>
+							<StatsPeriodNavigation date={ date } period={ period } url={ pathTemplate }>
+								<DatePicker period={ period } date={ date } showQueryDate />
+							</StatsPeriodNavigation>
+							<Intervals selected={ period } pathTemplate={ pathTemplate } compact={ true } />
+						</StatsPeriodHeader>
+					</div>
```
Comment (jsnmoon) [inline client/my-sites/stats/subscribers-section/index.tsx]: Let's avoid adding unnecessary wrapping div's in the future.

**Review by jsnmoon — DISMISSED** [review]
There are a few problems I recommend we fix before shipping this:

image

- The heading provided by `StatsPeriodNavigation` is unnecessary. The only components we need are the arrow components.
- This PR inadvertently reverts most of #75653.

I'll push up a commit that should remedy these issues.

**Comment by jsnmoon** [comment]
I pushed a commit to fix the unintentional revert and remove the unnecessary double heading. It does, however, remove the left/right arrows. Let's work on restoring those in a follow-up PR.

The subscribers chart section now looks like this:

image

File: client/my-sites/stats/subscribers-section/index.tsx [hunk 7]
```diff
@@ -32,22 +34,21 @@ function transformData( data: SubscribersData[] ): uPlot.AlignedData {
...
-	const period = 'month';
 	const quantity = 30;
 	const {
 		isLoading,
 		isError,
 		data,
 		// error,
 		status,
-	}: UseQueryResult< SubscribersDataResult > = useSubscribersQuery( siteId, period, quantity );
+	}: UseQueryResult< SubscribersDataResult > = useSubscribersQuery( period, quantity );
```
Comment (jsnmoon) [inline client/my-sites/stats/subscribers-section/index.tsx:51]: Why was `siteId` removed from this component? Wouldn't this hook still need the site ID to correctly query for this data?

**Pull request merged**